app/src/main/java/com/android/wildex/model/cache/usersettings/UserSettingsDataStore.kt

```diff
@@ -0,0 +1,54 @@
+package com.android.wildex.model.cache.usersettings
+
+import android.content.Context
+import androidx.datastore.preferences.core.Preferences
+import androidx.datastore.preferences.core.booleanPreferencesKey
+import androidx.datastore.preferences.core.edit
+import androidx.datastore.preferences.core.intPreferencesKey
+import androidx.datastore.preferences.preferencesDataStore
+import com.android.wildex.model.user.AppearanceMode
+import kotlinx.coroutines.flow.first
+
+val Context.userSettingsDataStore by preferencesDataStore(name = "user_settings_cache")
+
+/** Preference keys for user settings. */
+object UserSettingsPreferencesKey {
+  val ENABLE_NOTIFICATIONS = booleanPreferencesKey("enable_notifications")
+  val APPEARANCE_MODE = intPreferencesKey("appearance_mode")
+}
```

Comment:
And use helper functions for the keys instead, since we need different ones for each user


```suggestion
  }
  
  private fun enableNotificationsKey(userId: Id) =
    booleanPreferencesKey("enable_notifications_$userId")

  private fun appearanceModeKey(userId: Id) =
    intPreferencesKey("appearance_mode_$userId")

```